controllers/controller/devworkspacerouting/solvers/common.go

```diff
@@ -220,7 +220,7 @@ func getIngressForEndpoint(routingSuffix string, endpoint controllerv1alpha1.End
 			Labels: map[string]string{
 				constants.DevWorkspaceIDLabel: meta.DevWorkspaceId,
 			},
-			Annotations: nginxIngressAnnotations(endpoint.Name),
+			Annotations: createAnnotations(endpoint.Name, annotations, nginxIngressAnnotations),
```

Comment:
I'm not sure about this approach; there's a pretty distant link between where `annotations` is defined and where it's used, and how it's defined completely changes the code significantly further down the call stack. The function `createAnnotations` behaves completely differently based on whether `annotations` is an empty map here:

```go
	if routingAnnotations == nil || len(routingAnnotations) == 0 {
		appendMap(annotations, defaultAnnotations)
	} else {
		appendMap(annotations, routingAnnotations)
	}
```